app/aboot/lk2nd-motorola.c

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include <debug.h>
+#include <libfdt.h>
+#include <malloc.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "lk2nd-device.h"
+#include "lk2nd-motorola.h"
+#include "smem.h"
+
+#define SMEM_ID_VENDOR0          0x86
+#define SMEM_KERNEL_RESERVE      SMEM_ID_VENDOR0
+#define SMEM_KERNEL_RESERVE_SIZE 1024
+
+static void fdt_getprop_u32(const void *fdt, int offset, const char *name, uint32_t *val) {
+	int len;
+	const uint32_t *prop = fdt_getprop(fdt, offset, name, &len);
+	if (len == sizeof(*val))
+		*val = fdt32_to_cpu(*prop);
+	else
+		dprintf(INFO, "Failed to get prop %s in device tree", name);
+}
+
+void lk2nd_motorola_smem_write_unit_info(const void *fdt, int offset)
+{
+	int len, ret = 0;
+	fdt_getprop(fdt, offset, "lk2nd,motorola-unitinfo", &len);
+	if (len == 0) {
+		return;
+	}
+
+	struct mmi_unit_info *unit_info;
+	unit_info = malloc(SMEM_KERNEL_RESERVE_SIZE);
+	memset(unit_info, 0, SMEM_KERNEL_RESERVE_SIZE);
+
+	int chosen_offset = fdt_path_offset(fdt, "/chosen");
+	if (chosen_offset < 0) {
+		dprintf(INFO, "Failed to get /chosen node in device tree");
+		return;
+	}
+
+	unit_info->version = MMI_UNIT_INFO_VER;
+
+	const char *baseband = fdt_getprop(fdt, chosen_offset, "mmi,baseband", &len);
+	if (len < 0) {
+		baseband = "";
+	}
+	strlcpy(unit_info->baseband, baseband, BASEBAND_MAX_LEN);
```

Comment:
Move declaration of `baseband` to top. Also, `unit_info->baseband` is empty by default since it's zeroed. Therefore you can change the if condition to `len > 0` and move the `strlcpy` into the if statement.

Also, we may read beyond the end of the property if the string is not null terminated for stupid reasons. I think you can avoid that using:

```c
strlcpy(unit_info->baseband, baseband, MIN(len, BASEBAND_MAX_LEN));
```

(note the `MIN`)